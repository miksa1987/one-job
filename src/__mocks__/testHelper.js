export const saveTodoMock = jest.fn();