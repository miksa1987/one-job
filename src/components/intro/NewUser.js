import React from 'react';
import Form from '../common/Form';
import useField from '../../hooks/useField';
import Firebase from '../../firebase/firebase';
import TodoStore from '../../store';

const NewUserform = (props) => {
  const [ email, setEmail ] = useField('text');
  const [ password, setPassword ] = useField('password');
  const [ repeatPassword, setRepeatPassWord ] = useField('password');

  const store = React.useContext(TodoStore);
  const firebase = React.useContext(Firebase);

  const createUser = async (event) => {
    event.preventDefault();
    const response = await firebase.createNewUser(email.value, password.value);

    const user = {
      uid: response.uid,
      email: response.email,
      refreshtoken: response.refreshToken
    };

    store.setUser(user);
    window.localStorage.setItem('onejob-user', JSON.stringify(user));

    setEmail('');
    setPassword('');
    setRepeatPassWord('');
    props.setView('buttons');
  }


  return (
    <div>
    <Form onSubmit={createUser}>
      <input id='email' placeholder='Your e-mail' {...email} />
      <input id='password' placeholder='Desired password' {...password} />
      <input id='repeat-password' placeholder='Repeat password' {...repeatPassword} />
      <button id='create-user' type='submit'>
        Create new user
      </button>
    </Form>
    </div>
  );
}

export default NewUserform;