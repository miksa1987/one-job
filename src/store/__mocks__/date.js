export default class DateHandlerMock {
  setDate = jest.fn();
  setNextDayDate = jest.fn();
  setPreviousDayDate = jest.fn();
  setCurrentDayDate = jest.fn();
}