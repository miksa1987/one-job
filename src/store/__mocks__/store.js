import { decorate, computed, observable } from 'mobx';
import { createContext } from 'react';
import { 
  saveTodoMock, 
  setCurrentDateMock,
  loginAndSetUserMock,
  createAndSetUserMock,
  setNotificationMock 
} from './testHelper';

class TodoStore {
  currentTodo = {
    task: 'test',
    reflect: 'test',
    date: '2019-11-13',
    time: [ 12, 0 ],
    key: '1'
  };
  currentDate = '2019-11-13';

  setCurrentDate = (date) => {
    setCurrentDateMock();
    this.currentDate = date;
  }

  saveTodo = (newTodo, key) => {
    saveTodoMock();
    this.currentTodo = { ...newTodo, key: '1' };
  }

  setNextDayDate = () => this.currentDate = '2019-11-14';
  setPreviousDayDate = () => this.currentDate = '2019-11-12';
  setCurrentDayDate = () => this.currentDate = '2019-11-13';
  
  seeIfTodoTimeIsPassed = () => {}
  setNotification = () => setNotificationMock();

  loginAndSetUser = () => loginAndSetUserMock();
  createAndSetUser = () => createAndSetUserMock();

  get user() {
    return {};
  }
  
  get todo() {
    return this.currentTodo;
  }

  get date() {
    return this.currentDate;
  }

  get allTodos() {
    return [ this.currentTodo, this.currentTodo, this.currentTodos ];
  }

  get todoTimePassed() {
    return true;
  }
}

decorate(TodoStore, {
  currentUser: observable,
  currentTodo: observable,
  currentDate: observable,
  todos: observable,
  user: computed,
  todo: computed,
  date: computed,
  allTodos: computed
});

export default createContext(new TodoStore());