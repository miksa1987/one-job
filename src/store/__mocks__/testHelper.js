export const saveTodoMock = jest.fn();
export const setCurrentDateMock = jest.fn();
export const createAndSetUserMock = jest.fn();
export const loginAndSetUserMock = jest.fn();
export const setNotificationMock = jest.fn();