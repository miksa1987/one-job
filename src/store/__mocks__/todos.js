export default class TodosHandlerMock {
  getAndSetTodos = jest.fn();
  setCurrentTodo = jest.fn();
  checkForTodo = jest.fn();
  saveTodo = jest.fn();
  seeIfTodoTimeIsPassed = jest.fn();
}