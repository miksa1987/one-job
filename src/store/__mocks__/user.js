export default class UserHandlerMock {
  createAndSetUser = jest.fn();
  loginAndSetUser = jest.fn();
  logoutUser = jest.fn();
  setUser = jest.fn();
}