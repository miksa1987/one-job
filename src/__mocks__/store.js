import { decorate, computed, observable } from 'mobx';
import { createContext } from 'react';
import { saveTodoMock } from './testHelper';

class TodoStore {
  currentTodo = {
    task: 'test',
    reflect: 'test',
    date: '2019-11-13',
    time: [ 12, 0 ],
    key: '1'
  };

  saveTodo = (newTodo, key) => {
    saveTodoMock();
    this.currentTodo = { ...newTodo, key };
  }

  get user() {
    return {};
  }
  
  get todo() {
    return this.currentTodo;
  }

  get date() {
    return '2019-11-13';
  }

  get allTodos() {
    return [ this.currentTodo, this.currentTodo, this.currentTodos ];
  }
}

decorate(TodoStore, {
  currentUser: observable,
  currentTodo: observable,
  currentDate: observable,
  todos: observable,
  user: computed,
  todo: computed,
  date: computed,
  allTodos: computed
});

export default createContext(new TodoStore());