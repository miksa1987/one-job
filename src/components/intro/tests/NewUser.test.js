import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NewUser from '../NewUser';
import { createAndSetUserMock } from '../../../store/__mocks__/testHelper';

jest.mock('../../../store/store');

describe('NewUser', () => {
  afterEach(cleanup);

  test('Rendering is ok', () => {
    const component = render(<NewUser />);
    
    expect(component).toMatchSnapshot();
  });

  test('Form submit calls right functions', () => {
    const setView = jest.fn();
    const component = render(<NewUser setView={setView} />);
    const createButton = component.container.querySelector('#create-user');

    fireEvent.click(createButton);

    setTimeout(() => {
      expect(createAndSetUserMock.mock.calls.length).toBe(1);
      expect(setView.mock.calls.length).toBe(1);
    }, 50);
  });
})