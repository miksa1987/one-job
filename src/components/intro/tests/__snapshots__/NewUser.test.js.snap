// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUser Rendering is ok 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-bdVaJa dkwtxO"
      >
        <input
          id="email"
          placeholder="Your e-mail"
          type="text"
          value=""
        />
        <input
          id="password"
          placeholder="Desired password"
          type="password"
          value=""
        />
        <input
          id="repeat-password"
          placeholder="Repeat password"
          type="password"
          value=""
        />
        <button
          id="create-user"
          type="submit"
        >
          Create new user
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-bdVaJa dkwtxO"
    >
      <input
        id="email"
        placeholder="Your e-mail"
        type="text"
        value=""
      />
      <input
        id="password"
        placeholder="Desired password"
        type="password"
        value=""
      />
      <input
        id="repeat-password"
        placeholder="Repeat password"
        type="password"
        value=""
      />
      <button
        id="create-user"
        type="submit"
      >
        Create new user
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
